packages/loader/container-loader/src/container.ts

```diff
@@ -471,12 +480,15 @@ export class Container
 			container.mc.logger,
 			{ eventName: "RehydrateDetachedFromSnapshot" },
 			async (_event) => {
-				const deserializedSummary = JSON.parse(snapshot);
-				if (!isCombinedAppAndProtocolSummary(deserializedSummary, hasBlobsSummaryTree)) {
+				const detachedContainerState = JSON.parse(snapshot);
```

Comment:
i made a comment below about moving logic of rehydrateDetachedFromSnapshotOld into a free function. i think we could also move the type guard checking, basically, have that free funtion take the raw pending state string, and return the IPendingDetachedContainerState, so it encapsulates all the back compat and parsing

Reply:
this will also help remove a bunch of the old logic spread about to deal with the old format.